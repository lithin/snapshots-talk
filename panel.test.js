import React from 'react';
import Panel from './panel';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Panel', () => {
  it('renders number, and three buttons', () => {
    const result = shallow(<Panel />);
    expect(toJson(result)).toMatchSnapshot();
  });

  it('updates the number when a button is clicked', () => {
    const result = mount(<Panel />);
    expect(result.find('span').text()).toEqual('0');

    result.find('button').first().simulate('click');

    expect(result.find('span').text()).toEqual('1');
  });
});
