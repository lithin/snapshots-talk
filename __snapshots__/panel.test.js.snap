// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders number, and three buttons 1`] = `
<div>
  <span>
    0
  </span>
  <Button
    onClick={[Function]}
    type="primary"
  >
    Up
  </Button>
  <Button
    onClick={[Function]}
    type="primary"
  >
    Down
  </Button>
  <Button
    onClick={[Function]}
    type="primary"
  >
    Zero
  </Button>
</div>
`;
